Supported AI models (#819)
SimpleDataLabsInc/prophecy-docs: docs/copilot/copilot.md

  - generativeai
---

import Mermaid from '@theme/Mermaid';

> This page provides an introduction to Prophecy's AI capabilities, including descriptions of its task-oriented and agentic features.

Reviewer: ```suggestion
This page introduces Prophecy's AI capabilities, including descriptions of its task-oriented and agentic features.
```

:::

### Supported providers

<Mermaid
value={`flowchart LR
subgraph L1["Layer 1 — Application"]
P["Prophecy"]
end

subgraph L2["Layer 2 — Model Providers"]
DOA["Direct OpenAI API"]
EOA["Enterprise OpenAI API"]
AOA["Azure OpenAI"]
DGA["Direct Gemini API"]
VA["Vertex AI"]
DAA["Direct Anthropic API"]
end

subgraph L3["Layer 3 — Model Families"]
GPT["OpenAI GPT models"]
GEM["Google Gemini models"]
CLAUDE["Anthropic Claude models"]
end

P --> DOA
P --> EOA
P --> AOA
P --> DGA
P --> VA
P --> DAA

DOA --> GPT
EOA --> GPT
AOA --> GPT
DGA --> GEM
VA --> GEM
DAA --> CLAUDE

%% Remove background styling from subgraphs
classDef subgraphStyle fill:transparent,stroke:#333,stroke-width:1px
class L1,L2,L3 subgraphStyle
`}
/>

### Supported models

While Prophecy can connect to all providers shown in the diagram, the following models are officially tested and supported:

- `gpt-4o`
- `gpt-4o-mini`
- `gemini-2.5-flash`
- `gemini-2.5-flash-lite`

## Security

Prophecy employs rigorous industry practices to safeguard the security of the Prophecy application and maintain the privacy of customer data. Below are just a few components of our comprehensive security strategy and system structure:

> - Prophecy **does not** store or send anything from your data plane to any third-party large language model (LLM) providers. Instead, Prophecy makes use of rich metadata to construct the knowledge graph. As a result, Prophecy can interface with LLM providers while maintaining the privacy of the data itself.

Reviewer: ```suggestion
- Prophecy **does not** store or send your data to any third-party large language model (LLM) providers. Instead, Prophecy uses rich metadata to construct its knowledge graph. As a result, Prophecy can interface with LLM providers while keeping your data private.
```

> - The Prophecy IDE is hosted on secure servers on AWS. All storage systems are encrypted, and all servers are tightly access controlled and audited. Metadata is encrypted in transit at all times..

Reviewer: ```suggestion
- Prophecy is hosted on secure servers running on AWS. All storage is encrypted, and all servers are tightly access controlled and audited. All metadata is encrypted in transit.
```

> - The Prophecy IDE accesses your environment through a single IP address dedicated to you, allowing you to protect access to your data resources at the network level. The credentials are stored per user, and only a fully authenticated user can access their environment.

Author: "The credentials are stored per user"

This isn't even true anymore, and this isn't related to AI, so I'm removing this bullet point.
